OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs

```diff
@@ -0,0 +1,391 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Adds a particle-based overlay.")]
+	public class WeatherOverlayInfo : ITraitInfo
+	{
+		[Desc("Factor for particle density. A smaller value increases the number of particles spawned.")]
+		public readonly int ParticleDensityFactor = 1311;
+
+		[Desc("The wind will change over time by the values of WindStrength, or just stick to the first value.")]
+		public readonly bool ChangingWind = true;
+
+		[Desc("Works only if ChangingWind is enabled. The values of the wind intensity and direction. ")]
+		public readonly int[] WindStrength = { -5, -3, -2, 0, 2, 3, 5 };
+
+		[Desc("Works only if ChangingWind is enabled. Min. and max. ticks needed to change the wind. ")]
+		public readonly int[] WindTick = { 150, 750 };
+
+		[Desc("The wind is fading softly from zero or instant.")]
+		public readonly bool InstantWindChanges = false;
+
+		[Desc("Particles are drawn with rectangles when enable, else with lines.")]
+		public readonly bool RectOrLine = true;
+
+		[Desc("Works only with rectangles enabled. Size min. and max. value.")]
+		public readonly int[] PaticleSize = { 1, 3 };
+
+		[Desc("Scatters falling direction on the x-axis . Scatter min. and max. value.")]
+		public readonly int[] DirectionScatter = { -1, 1 };
+
+		[Desc("The value that particles fall down. Gravity min. and max. value.")]
+		public readonly int[] Gravity = { 100, 200 };
+
+		[Desc("The current offset value for the swing movement. SwingOffset min. and max. value.")]
+		public readonly int[] SwingOffset = { 10, 15 };
+
+		[Desc("The value that particles swing to the side each update. SwingSpeed min. and max. value.")]
+		public readonly int[] SwingSpeed = { 1, 25 };
+
+		[Desc("The value range that can be swung to the left or right. SwingAmplitude min. and max. value.")]
+		public readonly int[] SwingAmplitude = { 10, 15 };
+
+		[Desc("The colors for the particles.",
+			"For snow and rain use the temperate.pal with an index of 128-131 and 164-167.",
+			"and for a sandstorm at Dune2k the d2k.pal with an index of 78, 82, 85, 89.")]
```

Comment:
I'm not sure if we want mod specific content in the `Desc`s.
We could document that e.g. on a wiki page or just give a map as example (as you actually already do).